### DISCUSSION TITLE
Default to `rebase -s` when dragging?

### DISCUSSION THREAD

**Discussion by ilyagr** [gulbanana/gg#3] [Ideas]
This is a little bike-sheddy, but I'd like to suggest changing one default behavior in `gg`, at least once `feat: alternate drag modes for copy/duplicate, maybe for rebase-all-descendants` is implemented.

IMO, it's more "in the spirit of" `jj` to rebase a commit with all the descendants, similarly to how amending a commit changes all the descendants. So, I'd make that the default, and would have the current behavior happen if Ctrl is pressed.

Like in Inkscape, the status bar could say "Press Ctrl to move only one commit" while dragging, to make this discoverable.

Thanks once more for a cool GUI!

(To be clear, I'm partially thinking out loud as I'm writing this; this is not something I think of as extremely important, more an expression of one thing that felt a bit wrong as I was playing with the GUI).

**Comment by gulbanana** [comment]
I do think we want this feature. there are a few ways to get it, including:
- make `-s`-equivalent the default. the reason i didn't do this so far is that the interaction paradigm i have in mind is that the UI lets you make many small changes rapidly, so that you can operate at high- or low-granularity depending on your use case. for example, if you want to reorder a couple of commits relative to each other (something the CLI does not have great support for) the current `-r`-equivalent mode is better, and you *can* rearrange a whole branch... but it's tedious, I admit.
- drag modes - as you spotted. i'm not quite sure how to implement this yet, but i think modifier keys would be the way to go if html will cooperate. mode-hints in the status bar are a good idea.
- multiselect! i'd like to implement various forms of extra selection, such as shift-click to select range and double-click on branches to select everything up to their merge. this would let you easily(?) pick a whole bunch of changes on which to operate.
- other..?

**Reply by gulbanana** [reply]
in general i would like to expose jj's "model" rather than imitate its UI precisely, but in many cases it will be preferable for things to work the same way as in the CLI. one example where i've gone back and forth a bit is the "reset author" checkbox which modifies the behaviour of the Describe command - obviously this is similar to `jj describe --reset-author`. i think it would actually be more "ggish" to make it a separate command, and that's how i had it originally... but it's a very *niche* command, and there was UI space anyway. bit of a toss-up!

**Comment by gulbanana** [comment]
Multiple selection is the answer to this feature for now. It doesn't cover *all* case of -s, but works well for common operations like "move this entire linear branch".
